superset-frontend/src/explore/components/controls/DateFilterControl/DateFilterControl.tsx

```diff
@@ -183,15 +183,25 @@ export default function DateFilterControl(props: DateFilterLabelProps) {
   const [evalResponse, setEvalResponse] = useState<string>(value);
 
   useEffect(() => {
-    fetchTimeRange(value, endpoints).then(({ value, error }) => {
-      if (error) {
-        setEvalResponse(error || '');
-        setValidTimeRange(false);
-      } else {
-        setActualTimeRange(value || '');
-        setValidTimeRange(true);
-      }
-    });
+    const valueToLower = value.toLowerCase();
+    if (
+      valueToLower.startsWith('last') ||
+      valueToLower.startsWith('next') ||
+      valueToLower.startsWith('previous')
+    ) {
+      setActualTimeRange(value);
```

Comment:
Should we translate the value displayed for label?

Reply:
good point, I will add t().

Reply:
I think if you add `t()` here the backend will receive the translated value, we probably need a map with `key:label` here

Reply:
value is a dynamic value, not very easy to create a Map handle this.

Reply:
But there are limited number of combinations, right?